docs/apis/migration-perm-guidance.md

```diff
@@ -0,0 +1,87 @@
+When migrating, we recommend that you have less than 5,000 unique scopes per library. 
+
+### Role assignments
+ 
+A role assignment is a mapping between a user, a SharePoint object (Web, list, file) and a role (Design, Full Control, Contribute, etc.). The role assignment ties together the role definition (permission level) with the specific user or group, and the scope that that permission level will be applied to (such as list, folder, item).
+
+There is a role assignment limit of 5,000 per security scope.
+
+### Item limits 
+
+When a library (or list) contains more than 100,000 items (files and folders or list items), you cannot break permissions inheritance on the list itself.  There is a limit of 100,000 items that can be updated or removed as a part of creating a new SharePoint security scope. 
+
+If you are migrating a structure that has more than 100,000 children (such as files, folders, lists, or other object types), you need to restructure the migration by importing security in multiple phases to avoid exceeding this limit. Any VROOM invite, REST share link, or any other permission-modifying function call will trigger an HTTP 429 throttle if the threshold is reached. Permissions will not be updated. Unlike other throttles, waiting and trying again will not resolve the situation as you have reached a hard limit of 100,000.
+To learn more about the service limits in SharePoint for Microsoft 365, see [SharePoint Limits](https://docs.microsoft.com/en-us/office365/servicedescriptions/sharepoint-online-service-description/sharepoint-online-limits#items-in-lists-and-libraries).
+
+## Folders containing fewer than 100,000 items
+
+If your library (or list) has fewer than 100,000 items, and you have less than 50,000 unique scopes, you can migrate using the migration API and apply unique scopes to create folders with less 100K items to break inheritance as needed.
+
+## Folders containing more than 100,000 items
+
+If you have a folder with more than 100,000 items, we recommend one of the following approaches. Determine how many items are in your source or root folder, including lists or other object types. Scan and determine which folder structures have greater than 100,000 items.
+
+### Method 1: Restructure the source layout
+The first option is to restructure your *source* layout.
+
+For example, divide a single folder of 250,000 items into four folders at the root so each folder has less than 100,000 items. There is still “room to grow”, presuming users will continue to add content and make changes here.
+
+Make sure that there are no more than 50,000 unique scopes in the structure: ideally less than 5,000. 
+
+In this example, at the source, break up the structure into four folders, A, B, C, and D, each having less than 100,000 items. Then perform the migration. See the illustration below.
+
+![Hierarchy ABCD](../images/hierarchy-abcd.png)
+ 
+>[!Note]
```

Comment:
Alert type should be all CAPS [per MSDocs](https://review.docs.microsoft.com/en-us/help/contribute/markdown-reference?branch=master#alerts-note-tip-important-caution-warning)